src/app/(profile)/profiles/[position]/page.tsx

```diff
@@ -0,0 +1,35 @@
+﻿import ProfileList from '@/app/(profile)/(components)/ProfileList/ProfileList'
+import {
+  filterByAvailability,
+  filterByFullName,
+  filterByLocation,
+  filterBySeniority,
+  filterByTechnology,
+} from '@/app/(profile)/(components)/ProfileList/filters'
+import { type JobOfferFiltersEnum } from '@/app/(profile)/types'
+import { getPublishedProfilesPayloadByPosition } from '@/backend/profile/profile.service'
+import Loader from '@/components/Loader/Loader'
+import { Suspense } from 'react'
+
+export default async function PositionPage({
+  searchParams,
+  params,
+}: {
+  searchParams: Record<JobOfferFiltersEnum, string>
+  params: { position: string }
+}) {
+  const profiles = await getPublishedProfilesPayloadByPosition(params.position)
+
+  const filteredProfiles = profiles
+    .filter(filterBySeniority(searchParams.seniority?.split(',')))
+    .filter(filterByLocation(searchParams.location?.split(',')))
+    .filter(filterByTechnology(searchParams.technology?.split(',')))
+    .filter(filterByAvailability(searchParams.availability?.split(',')))
+    .filter(filterByFullName(searchParams.search))
+
+  return (
+    <Suspense key={JSON.stringify(searchParams)} fallback={<Loader />}>
```

Comment:
Why such key?

Reply:
Referring to this part of docs: [https://react.dev/reference/react/Suspense#resetting-suspense-boundaries-on-navigation](url)
Next does not recognize search params changing as a route changing so it does not retrigger the suspense fallback but this solution was for case without having filters inside `layout.tsx`. Now, I can even remove all suspense boundary from the pages 😁